### PR TITLE
Upgrade to zetasql-2024-03-01 and bump deps

### PR DESCRIPTION
**Pull request by hamnis**
Trying to move this forward by upgrading dependencies

### CODE REVIEW

File: zetasql-toolkit-bigquery/src/main/java/com/google/zetasql/toolkit/options/BigQueryLanguageOptions.java [hunk 1]
```diff
@@ -25,150 +25,197 @@
...
+                        LanguageFeature.FEATURE_V_1_3_PIVOT,
+                        LanguageFeature.FEATURE_V_1_3_QUALIFY,
+                        LanguageFeature.FEATURE_V_1_3_REMOTE_FUNCTION,
+                        LanguageFeature.FEATURE_V_1_3_REPEAT,
+                        LanguageFeature.FEATURE_V_1_3_SCRIPT_LABEL,
+                        LanguageFeature.FEATURE_V_1_3_UNNEST_AND_FLATTEN_ARRAYS,
+                        LanguageFeature.FEATURE_V_1_3_UNPIVOT,
+                        LanguageFeature.FEATURE_V_1_3_WITH_GROUP_ROWS,
+                        LanguageFeature.FEATURE_V_1_3_WITH_RECURSIVE,
+                        LanguageFeature.FEATURE_V_1_4_BARE_ARRAY_ACCESS,
```
Comment (ppaglilla) [inline zetasql-toolkit-bigquery/src/main/java/com/google/zetasql/toolkit/options/BigQueryLanguageOptions.java:118]: We should not enable all version 1.4 language features here, only those that BigQuery has. For this PR I'd only add the following, which are the ones I'm absolutely sure BigQuery supports.

This list misses some features which are supported for sure, feel free to add any that you know of. We should closely look at each one and enable them accordingly, but I'd keep that to a separate PR. 

* `FEATURE_V_1_4_ARRAY_AGGREGATION_FUNCTIONS`
* `FEATURE_V_1_4_REMOTE_MODEL`
* `FEATURE_V_1_4_GROUPING_BUILTIN`
* `FEATURE_V_1_4_GROUPING_SETS`
* `FEATURE_V_1_4_GROUP_BY_ALL`

**Review by ppaglilla — CHANGES_REQUESTED** [review]
Thanks for contributing this! I requested a couple of small changes.

Once we merge this, I'll immediately release v0.5.0. It has been a long time coming 😅.

**Comment by ppaglilla** [comment]
I took the liberty of applying the couple of changes I requested myself and merging. I hope that's not an issue. Thanks again for the contribution!

I should be able to release 0.5.0 with these changes shortly.

**Comment by hamnis** [comment]
Great. I was just heading in to work now to get this done, so this is perfect. Thanks a lot

**Pull request merged**